bindings/java/mac_run.me

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh -e
+
+cd `dirname $0`
+
+#unset _JAVA_OPTIONS
+if [ "x$JAVA_HOME" = "x" ]; then
+    JAVA_HOME=`/usr/libexec/java_home`
+fi
+
+if [ ! -f ../libblst.a -o ../blst.h -nt ../libblst.a ]; then
+    (cd ..; ../build.sh)
+fi
+
+if [ ! -f blst_wrap.cpp -o ../blst.swg -nt blst_wrap.cpp \
+                        -o ../blst.hpp -nt blst_wrap.cpp \
+                        -o ../blst.h   -nt blst_wrap.cpp ]; then
+    (set -x; swig -c++ -java -outdir . -o blst_wrap.cpp ../blst.swg)
+fi
+
+if [ ! -f libblst.dylib -o blst_wrap.cpp -nt libblst.dylib \
+                     -o ../libblst.a  -nt libblst.dylib ]; then
+    (set -x; c++ -march=native -shared -o libblst.dylib -fPIC \
+                 -I${JAVA_HOME}/include -I${JAVA_HOME}/include/darwin -I.. \
+                 -O -Wall -Wno-unused-function blst_wrap.cpp \
+                 ../libblst.a -Wl)
+fi
+
+if [ ! -f blst.class -o blst.java     -nt blst.class \
+                     -o runnable.java -nt runnable.class ]; then
+    (set -x; $JAVA_HOME/bin/javac *.java)
+fi
+
+env LD_LIBRARY_PATH=. $JAVA_HOME/bin/java runnable
```

Comment:
So far I believed that it's called `DYLD_LIBRARY_PATH` om macos. Does LD_LIBRARY_PATH actually work? Well, it's mostly for educational purposes, as I figured that it's possible to ask `java` to do whatever necessary to find library in current working directory on current OS with `-Djava.library.path=.`